### ISSUE TITLE
Persisting entities with detached objects

### ISSUE THREAD

**Issue by Xset-s** [hibernate/hibernate-reactive#353]
Hello again, first of all, thank you so much for your quick response to my issues. Everything works well and i m so glad about that.
But, as far as i'm concerned, for now (after recent commits) its impossible to persist objects with detached entities within.

I mean something like that:

```
final SampleEntity sampleEntity = new SampleEntity();

...

factory.withTransaction((session, transaction) -> session.persist(sampleEntity))
		.await().indefinitely();
```

and then

```
final SampleJoinEntity sampleJoinEntity = new SampleJoinEntity();
sampleJoinEntity.sampleEntity = sampleEntity; // sampleEntity is detached here

...

factory.withTransaction((session, transaction) -> session.persist(sampleJoinEntity))
		.await().indefinitely();
```

After executing ```session.persist(sampleJoinEntity)``` hibernate throws ```object references an unsaved transient instance - save the transient instance before flushing``` (but its not quite transient, its detached 🤔)

Yep, to achive expected functionality, i should use merge:

```
factory.withTransaction((session, transaction) ->
	session.merge(sampleEntity).invokeUni((aSampleEntity) -> {
		final SampleJoinEntity sampleJoinEntity = new SampleJoinEntity();

		sampleJoinEntity.sampleEntity = aSampleEntity;

		return session.persist(sampleJoinEntity);
	})).await().indefinitely();
```

Everything works fine, but in my view, its quite expensive for performance to merge all the necessary entities before persisting. 
Btw some commits ago i was able to persist objects with detached instances within. 

So is that a temporary restriction in your libary? Or maybe i did something wrong and there is a correct way to implement such persisting?

**Comment by gavinking** [comment]
This is a regression. I'm already fixing it.

**Comment by gavinking** [comment]
This is fixed in #354.
